src/components/ContributionButtonWithReminder.tsx

```diff
@@ -93,60 +106,60 @@ const RemindMeButton = ({ remindMeButtonText, onClick }: RemindMeButtonProps) =>
 interface ContributionButtonWithReminderProps {
     buttonText: string;
     buttonUrl: string;
-    remindMeButtonText?: string;
-    remindMeConfirmationText?: string;
-    remindMeConfirmationHeaderText?: string;
+    reminderStage?: ReminderStage;
+    reminderOption?: string;
     trackClick: (buttonId: number) => void;
     hidePaymentIcons?: string;
 }
-type SectionState = 'DEFAULT' | 'REMINDER_CONFIRMED' | 'REMINDER_CONFIRMATION_CLOSED';
```

Comment:
I really liked having this as a single authority on the different states you can be in regarding reminders in this component. Now I think it's spread across two states: `reminderStatus` (which has type `ReminderStatus`) and also `reminderConfirmationOpen` (which is a boolean). I wonder if it's be worth combining them? Could `reminderConfirmationOpen` be represented as a new member of the `ReminderStatus` enum perhaps?

It feels like spreading across two states opens the possibility of representing impossible situations. For example what would it mean for `reminderConfirmationOpen` to be `true`, but `reminderStatus` to be `Editing`? It feels like there are combinations which should ever happen, so [it'd be nice to make them impossible to represent](https://www.youtube.com/watch?v=IcgmSRJHu_8).

Reply:
The reminder CTA functionality in RRCP Banner/Epic is messy and (I think) overcomplicated for our needs. This is because in RRCP the button has to cope with both signed-in users where the code fetches the user's email address using the `fetchEmail()` function supplied by DCR, and signed-out users where clicking the button opens a microform where users can manually enter their email which then needs to be sanitised etc.

We only need to handle signed-in users, so should be able to use `fetchEmail()` once we get DCR to pipe it into our Epic (and Banner).

I really like the way the Braze Epic/Banner newsletter signup CTA button works (with the animated dots) - if we use a similar approach here, then when the user clicks the Remind Me Later button they will see the animated dots, then a success/fail message will display in place of all the CTA buttons. I'll try to build something for us to look at in a new PR

Reply:
The user journey has changed significantly - see screenshots above